From 7b35155128e2b8399776617681dc51a49f9d3ae0 Mon Sep 17 00:00:00 2001
From: Super User <root@localhost.localdomain>
Date: Sat, 16 May 2026 10:14:11 +0800
Subject: [PATCH] KVM: arm64: HDBSS should not apply DBM to VFIO MMIO regions

HDBSS incorrectly adds DBM to VFIO MMIO BAR regions.
DBM should only be used for RAM, not device memory.

Fix: Add !device check before setting DBM bit.

Signed-off-by: OpenEuler Kernel Team <kernel@openeuler.org>
---
 arch/arm64/kvm/mmu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/kvm/mmu.c b/arch/arm64/kvm/mmu.c
index fbc62b27adc8..b505936a2f64 100644
--- a/arch/arm64/kvm/mmu.c
+++ b/arch/arm64/kvm/mmu.c
@@ -1679,7 +1679,7 @@ static int user_mem_abort(struct kvm_vcpu *vcpu, phys_addr_t fault_ipa,
 		prot |= KVM_PGTABLE_PROT_W;
 
 #ifdef CONFIG_ARM64_HDBSS
-	if (kvm->enable_hdbss && logging_active)
+	if (!device && kvm->enable_hdbss && logging_active)
 		prot |= KVM_PGTABLE_PROT_DBM;
 #endif
 
-- 
2.43.0

